### ISSUE TITLE
MicroG acts as if I haven't added my Google account.

### ISSUE THREAD

**Issue by rokejulianlockhart** [microg/GmsCore#1847]
**Describe the bug**
[I am unable to proceed past the add account screen even after properly authenticating. It acts as if I never added an account to MicroG. When I try again, I get the “This account already exists on your device.”](https://github.com/microg/GmsCore/issues/1403#issue-814126841)

**To Reproduce**
(Please wait a moment for some demonstration to be uploaded – I appear to need my PC to upload MP4 files to GitHub 🤷 – but it doesn't really require explanation.)

**Expected behavior**
All software should automatically detect the present account, and consequently not necessitate manual authentication.

**Screenshots**
![Screenshot_20221212-165921](https://user-images.githubusercontent.com/42837531/207107037-e98d6db2-678b-496a-9f91-c9a0015d8c47.png)
![Screenshot_20221212-165933](https://user-images.githubusercontent.com/42837531/207107085-14236392-70f9-453b-8364-62ecb5039fed.png)
![Screenshot_20221212-165943](https://user-images.githubusercontent.com/42837531/207107090-34b62442-dcdc-4d90-a9fe-26d08ae0eb87.png)

**System**
Android Version: 13
Custom ROM: CalyxOS 4.4.0 (TQ1A.221205.011.22404000)

**Additional context**
https://github.com/microg/GmsCore/issues/1403#issuecomment-1346588225

**Comment by TheHighGroundWins** [comment]
The add account button on microg just adds an account wether or not it already exists. If your phone says your account is there then it's there.

If your account doesn't show in google apps, then open something like google drive that detects your google account then go to your profile and manage google account. There you should see options to allow google apps to see your google account. Enable that and you should be good to go.

**Comment by ildar** [comment]
Manually allow the apps in trouble to read Contacts. (workaround)

**Comment by rokejulianlockhart** [comment]
@ildar, that doesn't allow Google Fit to work, because it doesn't request that permission.

**Comment by rokejulianlockhart** [comment]
@TheHighGroundWins, the screenshots posted demonstrate that I have already enabled those options. That was why I filed this report – very few applications are able to access my Google account despite that being enabled.

**Comment by ale5000-git** [comment]
@rokejulianlockhart: Please specify all apps where you have this issue.

**Comment by beedell-roke** [comment]
@ale5000-git, the undermentioned do not authenticate:

1. [`com.chrome.canary`](https://play.google.com/store/apps/details?id=com.chrome.canary)

1. [`com.google.android.apps.fitness`](https://play.google.com/store/apps/details?id=com.google.android.apps.fitness)

1. [`com.google.android.apps.walletnfcrel`](https://play.google.com/store/apps/details?id=com.google.android.apps.walletnfcrel)

1. [`com.google.android.apps.recorder`](https://play.google.com/store/apps/details?id=com.google.android.apps.recorder)

1. [`ninja.sesame.app.edge`](https://play.google.com/store/apps/details?id=ninja.sesame.app.edge)

1. [`com.shazam.android`](https://play.google.com/store/apps/details?id=com.shazam.android)

Additionally, [`com.google.android.apps.googleassistant`](https://play.google.com/store/apps/details?id=com.google.android.apps.googleassistant) simply doesn't work even when [`com.google.android.googlequicksearchbox`](https://play.google.com/store/apps/details?id=com.google.android.googlequicksearchbox) has been installed.

Lastly, the undermentioned do not even invoke:

1. [`com.google.android.apps.healthdata`](https://play.google.com/store/apps/details?id=com.google.android.apps.healthdata)

1. [`com.google.android.apps.safetyhub`](https://play.google.com/store/apps/details?id=com.google.android.apps.safetyhub)

1. [`com.google.android.accessibility.soundamplifier`](https://play.google.com/store/apps/details?id=com.google.android.accessibility.soundamplifier)

1. [`com.google.android.googlequicksearchbox`](https://play.google.com/store/apps/details?id=com.google.android.googlequicksearchbox)

**Comment by TheHighGroundWins** [comment]
@rokejulianlockhart sorry man theres nothing more I can do. maybe the app is more reliant google services that microg doesn't have. Google contacts wouldn't sync for me and I had to manually import my contacts. It seems that your account is detected but just not connecting to be online.

**Comment by beedell-roke** [comment]
Indeed, contacts synchronize via [`apkmirror.com/apk/google-inc/google-contacts-sync`](https://www.apkmirror.com/apk/google-inc/google-contacts-sync), so that problem is mostly separate, although I am surprised that the relevant APK is not provided by MicroG, as I am surprised that the Trichrome library is not.

The rest of the problems are, however, severe bugs in MicroG, and why I never choose it over a 1st-party implementation of the Play Services Framework if possible.

**Comment by TheHighGroundWins** [comment]
I tried google contacts sync before but it didn't work maybe that's just me.

I understand why microg is like this tho. It's basically porting google features without using closed source stuff. Reminds me of wine, like how if you want a feature you need to build something that acts as an interface.

Still waiting for nearby share to become a thing on microg. I'd like to work on it myself but it seems too advanced for me to even know where to start with it.

**Comment by ildar** [comment]
Answered many times: GoogleContactsSyncAdapter works only when installed as
a system app (best into /system/priv-app/).

And you still may consider alternatives: NextCloud or any other CardDav
server.

**Comment by Sapiosenses** [comment]
> @ildar, that doesn't allow Google Fit to work, not least because it doesn't request that permission.

As far as I know, Google Fitness (same as Google Fit?) was never supported on microG, this is known and intentional. 

It's possible it may be implemented at some time in the future, cannot say.

If you want to see which GMS/Gplay APIs are working, not working or will never work (eg because they are needlessly snoopery and the project founder doesn't want to facilitate that), please see the following page:


https://github.com/microg/GmsCore/wiki/Implementation-Status

**Comment by Sense101** [comment]
> I tried google contacts sync before but it didn't work maybe that's just me.
> 
> I understand why microg is like this tho. It's basically porting google features without using closed source stuff. Reminds me of wine, like how if you want a feature you need to build something that acts as an interface.
> 
> Still waiting for nearby share to become a thing on microg. I'd like to work on it myself but it seems too advanced for me to even know where to start with it.

I'd also like to work on fixing google contacts sync. From what I can see in the code, the method to sync contacts has simply not been implemented, which is a bit ridiculous. 

I'm sure there are other people out there who'd like to fix these issues so they can use microg properly. But in order for anyone else to make changes like that they would need to understand how the code works! And there is no guide or anything on this.

I would be more than happy to work on adding these missing methods to microg. But I (and probably others) just don't know where to start.

**Comment by Iey4iej3** [comment]
Right now, you might use an FOSS alternative as well, such as [DAVx5](https://www.davx5.com/tested-with/google/).

**Comment by trunksn1** [comment]
I had the same problem then i found out how to solve it, it's easy.

1) put your account into revanced once

2) go to settings -> applications -> microG

3) select storage space, here press "delete cache" and then "delete data".

Go back to revanced and your account will be there waiting for you.

**Comment by trunksn1** [comment]
I stumbled upon this issue while having the same exact problem trying to sign in on ReVanced.

To solve that i just went into settings -> applications -> microG -> storage space and clicked on the two buttons to empty the cache and the data.


Hope this can help

**Comment by george2781** [comment]
So this is still very much an issue and no recommended fix in this issue either works or is feasible for me since I fully need G-suite apps to do college work, specifically I need Google Drive and Google Classroom to fully function.

Google Drive shows the exact behavior the issue describes, even with every single toggle I could find about letting apps do things turned on.

Google Classroom opens an accounts dialog that has no accounts in it, attempting to add my account will have microg tell me it's already added.

This is with the contacts sync adapter and calendar sync adapter installed as system apps and (provably) fully functional.

I hope this is fixed soon because this is a pretty big deal breaker for me.

**Comment by mar-v-in** [comment]
@george2781 you are probably running into issue #2680 right now. Give it a few days/weeks for ROMs to update accordingly (some will be faster and some slower depends on what you're using)

**Comment by litoj** [comment]
I have a similar (/same?) issue that <https://play.google.com/store/apps/details?id=cz.mafra.jizdnirady> has a google login method that shows me the microg account selector. Although it is populated with the one account I have logged in, selecting it just closes the dialog and the requesting app prints an error translatable as "Connecting via external service failed.

**Comment by ildar** [comment]
@litoj , per s.e. it's a different issue.

**Comment by Aquind** [comment]
I'm having the same issue with WhatsApp and microG 0.3.9 in /e/OS 3.2 (android 14)

Login Loop and then getting the, "Account already exists on this device" when trying to connect to a google account.

"allow apps to find accounts", "trust google for app permissions", and "Authenticate with device registration" are all enabled.

**Comment by LuNeder** [comment]
Same issue on WhatsApp when trying to enable backups (no degoogled way to back this up apparently, seedvault always errors on whatsapp)

Added the account first from YouTube, enabled all microg settings to allow apps to see the account, but WhatsApp still tries to make me add a new google account and then I get the "already on device" message

microg 0.3.13.250932-27 
/e/OS  3.7.3

**Comment by beedell-roke** [comment]
> No de-Google'd way to back this up, apparently: SeedVault always errors on WhatsApp.

@LuNeder, have you reported that to SeedVault, too?

**Comment by LuNeder** [comment]
> > No de-Google'd way to back this up, apparently: SeedVault always errors on WhatsApp.
> 
> [@LuNeder](https://github.com/LuNeder), have you reported that to SeedVault, too?

It's a known issue as WhatsApp doesn't use the normal android backup api or something. You can manually select its folder with folder backup on seedvault, which I'm doing, but I've seen conflicting information about whether restoring that still works.

I do wonder if I remove the google account and re-login from the WhatsApp button YouTube will also be unable to find the account (tho I don't really have storage on google so I don't think the backup would fit anyway actually...)
